src/maker.rs

```diff
@@ -441,40 +453,36 @@ mod tests {
     #[test]
     fn rate_updated_if_rate_update_with_same_value() {
         let mut maker = Maker {
-            mid_market_rate: MidMarketRate {
+            mid_market_rate: Some(MidMarketRate {
                 value: Rate::try_from(1.0).unwrap(),
-                ..Default::default()
-            },
+            }),
             ..Default::default()
         };
 
         let new_mid_market_rate = MidMarketRate {
             value: Rate::try_from(1.0).unwrap(),
-            timestamp: Utc::now(),
         };
 
         let reaction = maker.update_rate(new_mid_market_rate).unwrap();
-        assert_eq!(reaction, RateUpdateDecision::RateUpdated);
-        assert_eq!(maker.mid_market_rate, new_mid_market_rate)
+        assert_eq!(reaction, RateUpdateDecision::NoRateChange);
+        assert_eq!(maker.mid_market_rate.unwrap(), new_mid_market_rate)
```

Comment:
:wrench: You can use `matches!` to avoid this unwrap.

Reply:
This might do: 

```
assert_eq!(maker.mid_market_rate, Some(new_mid_market_rate))
```

Matches is probably an overkill here :)